great_expectations/experimental/metric_repository/metric_list_retriever.py

```diff
@@ -0,0 +1,322 @@
+        table_column_types = list(
+            filter(
+                lambda m: m.metric_name == MetricTypes.TABLE_COLUMN_TYPES, table_metrics
+            )
+        )[0]
+
+        # We need to skip columns that do not report a type, because the metric computation
+        # to determine semantic type will fail.
+        exclude_column_names = self._get_columns_to_exclude(table_column_types)
+
+        numeric_column_names = self._get_numeric_column_names(
+            batch_request=batch_request, exclude_column_names=exclude_column_names
+        )
+        timestamp_column_names = self._get_timestamp_column_names(
+            batch_request=batch_request, exclude_column_names=exclude_column_names
+        )
+        numeric_column_metrics = self._get_numeric_column_metrics(
+            metric_list, batch_request, numeric_column_names
+        )
+        timestamp_column_metrics = self._get_timestamp_column_metrics(
+            metric_list, batch_request, timestamp_column_names
+        )
+        all_column_names: List[str] = self._get_all_column_names(table_metrics)
+        non_numeric_column_metrics = self._get_non_numeric_column_metrics(
+            metric_list, batch_request, all_column_names
+        )
+
+        bundled_list = list(
+            chain(
+                table_metrics,
+                numeric_column_metrics,
+                timestamp_column_metrics,
+                non_numeric_column_metrics,
+            )
+        )
+
+        return bundled_list
+
+    def _get_non_numeric_column_metrics(
+        self,
+        metrics_list: List[MetricTypes],
+        batch_request: BatchRequest,
+        column_list: List[str],
+    ) -> Sequence[Metric]:
+        column_metric_names = {MetricTypes.COLUMN_NULL_COUNT}
+        metrics: list[Metric] = []
+        metrics_list_as_set = set(metrics_list)
+        metrics_to_calculate = sorted(
+            column_metric_names.intersection(metrics_list_as_set)
+        )
+
+        if not metrics_to_calculate:
+            return metrics
+
+        column_metric_configs = self._generate_column_metric_configurations(
+            column_list, list(metrics_to_calculate)
+        )
+        batch_id, computed_metrics, aborted_metrics = self._compute_metrics(
+            batch_request, column_metric_configs
+        )
+
+        # Convert computed_metrics
+        ColumnMetric.update_forward_refs()
+        metric_lookup_key: _MetricKey
+
+        for metric_name in metrics_to_calculate:
+            for column in column_list:
+                metric_lookup_key = (metric_name, f"column={column}", tuple())
+                value, exception = self._get_metric_from_computed_metrics(
+                    metric_name=metric_name,
+                    metric_lookup_key=metric_lookup_key,
+                    computed_metrics=computed_metrics,
+                    aborted_metrics=aborted_metrics,
+                )
+                metrics.append(
+                    ColumnMetric[int](
```

Comment:
The type provided here (`int`) would have to change depending on the type of metric. I think this method should not take the `metrics_list` parameter.